runtime/src/main/cpp/Exceptions.cpp

```diff
@@ -19,6 +19,8 @@
 #include <stdint.h>
 
 #include <exception>
+#include <atomic>
```

Comment:
This is C++11 which seems to be supported everywhere. And I have concerns regarding Atomic.h: is reimplementing std well justified in that case? Could you provide rationale to reinvent it? std has clearer interface and more features, also optional memory order parameter for the fine tuning. BTW code size is exactly the same when using `std::test_and_set` or  own `compareAndSet`, but the standard one looks better.

Reply:
In `Atomic.h` we use `KONAN_NO_THREADS` to disable atomics on the platforms that do not support threads. On these platforms I'm not sure that `<atomic>` will work at all.

Also, I do agree that using std interfaces is better, and we should consider mirroring them. So that with `KONAN_NO_THREADS` we implement the interfaces manually without atomicity and without `KONAN_NO_THREADS` we just `using` or manually forward them to stdlib implementation.

Reply:
Good point, got it. 
BTW c++-ish implementation of threading policy may simply replace atomic counters and flags with plain integer. Basically, atomicity w.r.t. a counter variable is a type quality, not the particular expression quality. Not sure if the memory barriers may be needed occasionally to prevent reordering as a result of compiler optimization or CPU pipeline rescheduling.